### PR TITLE
MDSAAS-361 TypeScript 3.7

### PR DESCRIPTION
**Pull request by drtyh2o**
[TypeScript 3.7](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-7.html) has landed 🚀 

If you use `VS Code`, you will need to explicitly tell it to _`Use Workspace Version`_ of TypeScript rather than the builtin version.

Screen Shot 2019-11-06 at 9 48 28 AM

Note that _`Prettier`_ does not yet understand the new `?.` and `??` syntax, but a new `1.19` release is expected today. That will be added to this PR prior to merging.

### CODE REVIEW

File: packages/mds-audit/types.ts [hunk 1]
```diff
@@ -16,13 +16,12 @@
 
 import { Audit, Telemetry, Timestamp, UUID } from '@mds-core/mds-types'
 import { ApiRequest, ApiResponse, ApiResponseLocals } from '@mds-core/mds-api-server'
+import { Params, ParamsDictionary } from 'express-serve-static-core'
 
 // Allow adding type definitions for Express Request objects
-export type AuditApiRequest = ApiRequest
+export type AuditApiRequest<P extends Params = ParamsDictionary> = ApiRequest<P>
```
[the remark this answers is not in the document]
Comment (drtyh2o) [reply packages/mds-audit/types.ts]: The `=` allows you to specify the default type for the generic. So `AuditApiRequest` and `AuditApiRequest<ParamsDictionary>` are equivalent.

**Comment by macsj200** [comment]
In #69 I'm using
`elem ?? 'nullish'`
Eslint is complaining about the question mark though
`Parsing error: Expression expected.`
Maybe I need to point eslint to the workspace version too?

**Comment by drtyh2o** [comment]
> In #69 I'm using
> `elem ?? 'nullish'`
> Eslint is complaining about the question mark though
> `Parsing error: Expression expected.`
> Maybe I need to point eslint to the workspace version too?

I haven't merged my PR yet (had to pull in other changes that got merged ahead of me) so you don't yet have latest eslint/parser.

**Comment by macsj200** [comment]
Hmm, I thought I had branched off the latest commit on this branch, but it looks like all I have is `d7dd64572c393e590d7ff5a88e521e532840a25a` which was the first commit. So I'd imagine a `rebase` would do the trick.

**Pull request merged**